x-pack/plugins/security/server/user_profile/user_setting_service.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { KibanaRequest } from '@kbn/core-http-server';
+import type { Logger } from '@kbn/logging';
+
+import type { UserProfileGetCurrentParams, UserProfileServiceStart } from './user_profile_service';
+
+export interface UserSettingServiceStart {
+  /**
+   * Returns the currently signed-in user's settings from their User Profile
+   *
+   * @param request the KibanaRequest that is required to get the current user and their settings
+   */
+  getCurrentUserProfileSettings(request: KibanaRequest): Promise<Record<string, string>>;
+}
+
+/**
+ * A service that wraps the {@link UserProfileServiceStart} so that only the 'getCurrent' method is made available
+ */
+export class UserSettingService {
+  private readonly logger: Logger;
+
+  constructor(logger: Logger) {
+    this.logger = logger;
+  }
+
+  start(userProfileServiceStart: UserProfileServiceStart): UserSettingServiceStart {
+    return {
+      getCurrentUserProfileSettings: async (request) => {
+        const params: UserProfileGetCurrentParams = {
+          request,
+          dataPath: '*',
```

Comment:
nit: if we are looking for `data.userSettings` in line 44. We may want to request only that `dataPath`.

Reply:
Great catch! Fixed in latest commit